pilot/pkg/serviceregistry/kube/controller/multicluster.go

```diff
@@ -155,8 +155,6 @@ func (m *Multicluster) ClusterAdded(cluster *multicluster.Cluster, clusterStopCh
 	// clusterStopCh is a channel that will be closed when this cluster removed.
 	options := m.opts
 	options.ClusterID = cluster.ID
-	// the aggregate registry's HasSynced will use the k8s controller's HasSynced, so we reference the same timeout
-	options.SyncTimeout = cluster.SyncTimeout
```

Comment:
I think we need to add this back, or make it

```
if !localCluster {
  options.SyncTimeout = features.RemoteClusterTimeout
}
```

I think the test still passes because the `pkg/kube/multicluster` blocks has synced, but the k8s controller doesn't.

Reply:
good catch, i removed the value setting, should add it here for non local cluster